src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -115,15 +115,14 @@ bool ShenandoahConcurrentGC::collect(GCCause::Cause cause) {
   {
     ShenandoahBreakpointMarkScope breakpoint_mark_scope(cause);
 
-    // Reset task queue stats here, rather than in mark_concurrent_roots
+    // Reset task queue stats here, rather than in mark_concurrent_roots,
     // because remembered set scan will `push` oops into the queues and
     // resetting after this happens will lose those counts.
     TASKQUEUE_STATS_ONLY(_mark.task_queues()->reset_taskqueue_stats());
 
     // Concurrent remembered set scanning
     entry_scan_remembered_set();
-    // When RS scanning yields, we will need a check_cancellation_and_abort()
-    // degeneration point here.
+    // TODO: When RS scanning yields, we will need a check_cancellation_and_abort() degeneration point here.
```

Comment:
We need to be really careful when transitioning to a degenerated cycle after RS scan is complete. Logically, the RS scan is part of the concurrent root scan, but it didn't get implemented with that pattern.

Reply:
Yes, I just converted this to TODO, because it looks like a TODO.

Reply:
@earthling-amzn : should this go into the "entry_mark_)roots()" below then? Clearly, it seems that we can't really have a degeneration point after remembered set scanning from how I understood your comment, so the documentation comment above (to which Aleksey merely added the `TODO:` prefix) is incorrect and should be either removed entirely or corrected suitably?

Reply:
I think it's fair to leave it as a TODO. Interrupting he RS scan is something we want and we have tried in the past, but we shelved it because it was more complicated than expected.